### PR TITLE
Implements UNIFIED_CREDIT_CODE_RANDOM_REPLACE mask algorithm

### PR DESCRIPTION
**Pull request by lushaorong**
ref #22759.

Changes proposed in this pull request:
  - Implements UNIFIED_CREDIT_CODE_RANDOM_REPLACE mask algorithm

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [x] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by zhaojinchao95** [comment]
@lushaorong Hi, there some conflict, please resolved it.

File: docs/document/content/user-manual/common-config/builtin-algorithm/mask.cn.md [hunk 1]
```diff
@@ -148,6 +148,18 @@ weight = 9
...
+
+类型：UNIFIED_CREDIT_CODE_RANDOM_REPLACE
+
+可配置属性：
+
+| *名称*            | *数据类型* | *说明*        |
+|-----------------|----------|-------------|
+| registration-department-codes | String   | 登记管理部门代码（以英文逗号分隔） |
+| category-codes | String   | 机构类别代码（以英文逗号分隔） |
+| administrative-division-codes | String   | 等级管理机关行政区划码（以英文逗号分隔） |
```
Comment (strongduanmu) [inline docs/document/content/user-manual/common-config/builtin-algorithm/mask.cn.md:161]: `行政区划随机码表` may be better.

**Comment by strongduanmu** [comment]
Hi @lushaorong, can you merge master branch?

**Review by strongduanmu — APPROVED** [review]
Good job, merged.

**Pull request merged**